tests/app/main/test_frameworks.py

```diff
@@ -2805,14 +2810,188 @@ def test_get_with_with_previous_answers(self, data_api_client):
             doc = html.fromstring(res.get_data(as_text=True))
             assert len(doc.xpath('//input[@id="input-PR1-no"]/@checked')) == 1
 
-    def test_post_valid_data(self, data_api_client):
+    def test_get_with_with_prefilled_answers(self, data_api_client):
         with self.app.test_client():
             self.login()
+            # First call is for the current framework; second call for the framework to pre-fill from
+            data_api_client.get_framework.side_effect = [
+                self.framework(slug='g-cloud-9', name='G-Cloud 9', status='open'),
+                self.framework(slug='digital-outcomes-and-specialists-2', name='Digital Stuff 2', status='live'),
+            ]
+            # Current framework application information
+            data_api_client.get_supplier_framework_info.return_value = self.supplier_framework(
+                framework_slug="g-cloud-9",
+                declaration={"status": "started"},
+                prefill_declaration_from_framework_slug="digital-outcomes-and-specialists-2"
+            )
+            # The previous declaration to prefill from
+            data_api_client.get_supplier_declaration.return_value = {
```

Comment:
I'm going to suggest leaving these functions alone for now. They're in the wrong place at the moment.
Happy to discuss these methods' inclusion and location another time.

I'd also say we don't strictly need to assert the calls in this test:
This test asserts the existence of elements and their state on the page. What this test asserts can be achieved entirely independently of the api calls.

If you want, on the other hand, to assert that a given piece of code makes certain calls then check the calls but not the html.

A caveat to this might be if you wanted to ensure that the state on the page can be achieved only with a certain number of api calls, however that's not what _this_ test is meant to do.


I'm not _that_ fussed if we check the calls here or not but I'd question wether or not it's causing fragility in the test suite by enforcing conditions we actually don't care about.